plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -426,6 +428,41 @@ else if (column.getBaseHiveColumnIndex() < fileColumns.size()) {
         }
     }
 
+    /**
+     * Recreate the list of fileColumns, updating the names of any whose names have changed in the
+     * corresponding elements of the desiredColumns list.  NOTE: this renaming is only applied to
+     * top-level columns, not nested columns.
+     * @param fileColumns All OrcColumns nested in the root column of the table.
+     * @param desiredColumns HiveColumnHandles for the metastore's table columns.
+     * @return Return the fileColumns list with any OrcColumn corresponding to a desiredColumn renamed if
+     * the names differ from those specified in the desiredColumns.
+     */
+    private static List<OrcColumn> ensureColumnNameConsistency(List<OrcColumn> fileColumns, List<HiveColumnHandle> desiredColumns)
+    {
+        int columnCount = fileColumns.size();
+        ImmutableList.Builder<OrcColumn> builder = ImmutableList.builderWithExpectedSize(columnCount);
+
+        Map<Integer, HiveColumnHandle> desiredColumnsByNumber = desiredColumns.stream()
+                .collect(toImmutableMap(HiveColumnHandle::getBaseHiveColumnIndex, identity()));
+
+        for (int index = 0; index < columnCount; index++) {
+            OrcColumn orcColumn = fileColumns.get(index);
```

Comment:
Could shorten the code a bit by naming this `column`

Reply:
Shortened.